src/components/LHNOptionsList/LHNOptionsList.tsx

```diff
@@ -1,9 +1,8 @@
 import {FlashList} from '@shopify/flash-list';
```

Comment:
@jbroma the changes in this file seems to have caused the issue below, only when a request is created offline


https://github.com/Expensify/App/assets/9680864/4e1e9416-ce5b-49fc-98ef-2246eaa8e6fe

Reply:
I think specifically the removal of `withCurrentReportID` HOC

Reply:
Actually it's the removal of `currentReportID` from `extraData` that's passed to `FlashList`. But I get it's necessary so the whole LHN doesn't re-render when we switch focus from a report, right? so how do we fix this, any ideas?

Reply:
but if we leave out from the dependency it will remain stale, it won't be update when we switch to another report, no?

Reply:
do you have focus mode enabled? if yes choose 'Recent' and try again. Also what updates? this PR on its own or what you mentioned?

Reply:
I don't think we want that though, if you remove it from the dependencies it will remain stale, the first value of it will remain in extra data even though we switched to another report

Reply:
But that will trigger a re-render of every LHN item, we just want to re-render the report we switched to I think and the one we switched from. Can you test your changes by adding `console.log` inside `renderItem` and see when it gets re-triggered?

Reply:
I think the question is why is the use of `currentReportID` in `OptionRowLHNData` not working as before when it was passed from as a prop to `LHNOptionsList`

Reply:
>It already does:

Yeah but here's the thing, `renderItem` in this PR is now only called one, and the re-rendering is offloaded to `OptionRowLHNData`, if you check out the production branch, you'll notice that `renderItem` gets called every time we switch to another report.

Reply:
> The useMemo gets triggered everytime we move from reports right?

Nope, it doesn't, only when those dependencies change and they don't when we're simply switching between reports.

Reply:
I can't reproduce this anymore for some reason @GandalfGwaihir, can you?

Reply:
nvm I just reproduced it again